### DISCUSSION TITLE
Recruitment

### DISCUSSION THREAD

**Discussion by e-perl-NOAA** [nmfs-stock-synthesis/stock-synthesis#566] [General]
Discussion for questions about recruitment. Please subscribe to this discussion so that you get notifications about discussion posts.

**Comment by e-perl-NOAA** [comment]
For recruitment, we *generally* advocate starting as early as catch becomes significant, not when other data start.

We also are increasingly seeing merit in using:
2 #do_recdev:  0=none; 1=devvector; 2=simple deviations
to give more flexibility to the recruitment time series, especially in cases where recruitment is not declining in accordance with a spawner-recruitment curve.

**Comment by e-perl-NOAA** [comment]
From [VLAB conversation about recdevs](https://vlab.noaa.gov/web/stock-synthesis/public-forums/-/message_boards/view_message/36560918):

The situation in which use of early recruitment devs is most useful is when:
- long time series of catch and sorter time series of data that are informative about recdevs
- desire to include uncertainty about early recdevs even if they cannot be estimated individually
- desire to use the recdev  option 1 for later recdevs so their mean value will closely match conditions of the estimated MSY

That lead to creation of the early recdevs which are a simple dev vector without a sum to zero constraint.  An advantage of use of early recdevs is that they can be turned on in a later phase to capture their variance without slightly extending runtime associated with estimating them in early phase.

Over time, we have learned, but not definitely demonstrated, that the option 1 dev-vector has disadvantages, particularly when doing MCMC, and that the sum-to-zero constraint can produce illogical patterns in the recdevs.

The current status is that I generally recommend using option 2 for all rec_devs and not using early recdevs.

**Reply by e-perl-NOAA** [reply]
From Adam Langley in response to the VLAB discussion:
I am currently undertaking two assessments and have been exploring a number of these issues. The species (NZ snapper) is relative long lived (M ~ 0.07). Both stocks have shown a marked increase in productivity over the last 10 years. The stocks have a good time series of age composition data from the mid 1970s onwards. There was a substantial amount of catch from the preceding decades although the magnitude of the catch is not well known.

Previously, we conducted a full catch history model starting from unexploited conditions. However, the change in productivity means that the B0 is no longer considered appropriate for management purposes. Starting the model under exploited conditions (in 1975) means that we no longer need to include the less reliable large catches from the earlier years. 

Previously, we have tended to estimate recruitments with the constrained dev vector (option 1). Some simulation work indicated that this approach had the potential to bias estimates of stock status. I also encountered issues with the stock crashing when running MCMC - when the stock was at very low biomass the constrained rec devs were unable to generate enough recruits (for a relatively small proportion of the MCMC draws, but frustrating). This was resolved when switching to unconstrained devs (option 2). Other comparisons did not result in appreciable differences in the estimated recruitment/biomass from the two options.

Starting the model in 1975, I am estimating an initial equilibrium fishing mortality - informed by the early age composition data. However, those data exhibit considerable variability in recruitment over the 20-30 y age range. So, I have also been estimating early rec devs to better fit those age comps and incorporate the uncertainty in the initial F. If those data aren't very informative then you can get confounding between the initial F and the early rec devs. The attached plot of the rec devs for the two periods (early and main) is attached. The very early (1960-65) recruitments are highly uncertain, while the 1970-74 deviates are quite well informed by the age comp data - there is good progression of some very strong year classes. Estimating the early rec devs allowed a good fit to those data, but didn't greatly change the estimate of initial F or initial biomass.  

I hope these observations are useful - this could serve as a potential case study.
<img src="https://github.com/nmfs-ost/ss3-source-code/assets/118312785/48ef9025-751f-408b-a64c-ea3ef9cf90d9" height="450">

**Comment by rpwildermuth-NOAA** [comment]
Hello,
I have a question about weighting user-provided recruitment deviations for the recent ("late") recruitment period. Our group is using an external model to fit a relationship between environmental drivers and recruitment deviations to provide an informed estimate of recruitment in the final assessment year and forecast year for Pacific sardine. I'm using the advanced recruitment 'read_recdevs' option to input the last two years of recruitment deviations into the assessment. If I leave the forecast recruitment phase = 0, the model shifts the estimated rec devs for these years away from my inputs, as if I didn't read them in. In [my current application](https://github.com/SWFSC/SardineRecruitIndex/blob/main/scenarioModels/benchmarkDFA_AsForecast/control.ss), I've changed the forecast recruitment phase = -1, which keeps the point estimate read into SS constant for these final years.

My question is: Is there a way to soften these read-in input rec devs so that they provide an informative prior for these final recruitment years? Alternatively, is there a way to change the weight given to the point estimates for these recruitment years so that the model isn't pinning the model fit to these specific inputs, but using them as buffers?

I've played with changing the last year of main rec devs and the 'end_yr_for_ramp_in_MPD', but maybe there's some combination of these recruitment settings that will get me somewhere closer to where I'd like to be.

Thank you ~ RW

**Comment by iantaylor-NOAA** [comment]
@rpwildermuth-NOAA, good question.

Instead of fixing the recent recdevs at the estimates from the analysis of environmental drivers, you could input the environmental driver as an index of recdevs. @okenk set up a Yellowtail Rockfish model in this way here https://github.com/pfmc-assessments/yellowtail_2025/tree/main/Model_Runs/5.04_ocean where some things to think about include
* use index units 36 to override selectivity and make link the index to recdevs, and
* for a short index, you may want to fix catchability (Q_base) at 1.0.

In general, the model has zero information about the final few recruitment deviations, so it may fit the index really well (depending on the index uncertainty relative to sigmaR).

Lastly, one additional control you have is the "Forecast Recruitment Deviations Lambda" described in the User Manual at https://nmfs-ost.github.io/ss3-doc/SS330_User_Manual_release.html#RecDevSetup as "This lambda is for the log likelihood of the forecast recruitment deviations that occur before endyr + 1. Use a larger value here if solitary, noisy data at end of time series cause unruly recruitment deviation estimation." That could be used to effectively scale the recent recdev estimates anywhere between the index observations and 0 (although changing the index uncertainty would do the same thing). That lambda won't impact fixed initial values, however.

**Reply by rpwildermuth-NOAA** [reply]
Thanks @iantaylor-NOAA,
We have a model version that applies this environmental index on rec devs "as data" (see [this model setup](https://github.com/SWFSC/SardineRecruitIndex/blob/rpwDev/R/fitDFAAsData.R)). Because this process generally isn't very informative for data rich assessments, we'd like to compare outcomes to the "as forecast" style described above.

I'll look into the forecast recruitment deviations lambda specifics and see if that might be a way to tune the information from the external environmental index. Sounds like it may take some fenagling with the other late/forecast rec dev specifications, but probably better than assuming point estimates. 
Thanks! ~ RW

**Reply by okenk** [reply]
Some random thoughts: 

Yes, if you want to "softly" inform the recdev a la a prior, I would use index type 36 and fix Q at 1. If you fix Q, even a single year of an index can be informative, including during the late recruitment deviation years. As Ian alluded to, the prior on recdevs pulling them to zero will compete with the index and whatever SE you give the index, but there is usually not any other "data" to cause a conflict in the final year, so the index will get fit pretty precisely unless it is more extreme, in which class the recdev will get pulled towards zero (a la 2021 and 2023 [here](https://github.com/pfmc-assessments/yellowtail_2025/blob/main/report/r4ss_plots/index2_cpuefit_SMURF.png)). For the forecast years, I'm not aware of a more probabilistic approach. Could you change your model end year?

**Reply by rpwildermuth-NOAA** [reply]
Ok, I think I'm understanding. So instead of providing the full index time series, just give it the final few years of data input then use a Q and Lambda to try and corral the estimate around the index. I can give that a try and then make sure the benchmark reference period matches the same "late" period distribution. That might get me closer to something I want.
